js/build.js

```diff
@@ -351,6 +428,174 @@
       }
     }
 
+    function requestPicture (fileInput) {
+      selectedFileInputName = fileInput.name;
+      var boundingClientRectTarget = fileInput;
+      var boundingRect = boundingClientRectTarget.getBoundingClientRect();
+      while( boundingRect.width === 0 || boundingRect.height === 0 ) {
+        if (!boundingClientRectTarget.parentNode) {
+          break;
+        }
+        boundingClientRectTarget = boundingClientRectTarget.parentNode;
+        boundingRect = boundingClientRectTarget.getBoundingClientRect();
+      }
+
+      return new Promise(function(resolve, reject){
+        var cameraOptions = {
+          boundingRect: boundingRect
+        };
+        navigator.notification.confirm(
+          'How do you want to choose your image?',
+          function onSelectedImageMethod (button) {
+            document.body.focus();
+            switch (button) {
+              case 1:
+                cameraOptions.source = Camera.PictureSourceType.CAMERA;
+                return resolve(cameraOptions);
+              case 2:
+                cameraOptions.source = Camera.PictureSourceType.PHOTOLIBRARY;
+                return resolve(cameraOptions);
+              default:
+                return;
+            }
+          },
+          'Choose Image',
+          ['Take Photo', 'Choose Existing Photo', 'Cancel']
+        );
+      });
+    }
+
+    function getPicture (options) {
+      options = options || {};
+      if (Fliplet.Env.get('platform') === 'web') {
+        return;
+      }
+
+      if (typeof navigator.camera === 'undefined') {
+        return Fliplet.Navigate.popup({
+          popupTitle: 'Sorry',
+          popupMessage: 'It looks like your app does not support file upload. Please contact the support team for more information.'
+        });
+      }
+
+      var popoverOptions = {
```

Comment:
I've trimmed all trailing whitespace and converted tabs to space. Text editor tells me there're no changes to save.